### PR TITLE
feat(query): capture queries

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

When debugging a user's session, I wished I could see all the queries performed in the replay timeline.

## Changes

Adds `posthog.capture` to `query()` in query.ts. This only captures the logged query, and the ClickHouse SQL translation if available (only for HogQLQuery queries for now). This does not capture any response, nor even the count of them.

The events captured are:

```ts
success = { event: 'Query', properties: { query: { ... }, duration: 1, clickhouse_sql: 'if available' }
failure = { event: 'Query Error', properties: { query: { ... }, duration: 1 }
```

Because we support JSON property access now, we can afterwards do stuff like:

```sql
select timestamp, properties.query.query, properties.duration from events where properties.query.kind = 'HogQLQuery' limit 100
```

## Questions

Are we exposing possible PII by logging the filters and hogql used in queries?

## How did you test this code?

Added a test.

### CODE REVIEW

**Comment by thmsobrmlr** [comment]
Filters often contain PII like emails. We're sending all this to ourselves though (where we already have the same data), so that should not be a problem (or I don't understand the question 😅 )

**Comment by pauldambra** [comment]
Does this overlap with the time to see data capture stuff? We should already be tagging this info in log comments and reporting it with time to see data.

For those we were also adding performance response headers that showed in the chrome network panel.

I like this idea of surfacing more performance in the replay listing generally but wonder if there's some tidying up needed too

(haven't read the code 🙈)

**Comment by pauldambra** [comment]
Screenshot 2023-05-25 at 10 31 52

Here's an example showing that loading a particular insight took no CH time and 22ms of PG time. (presumably it came from the cache and we could add server timings for that)

**Comment by pauldambra** [comment]
And you can find that query metrics with `select type, insight, insights_fetched, insights_fetched_cached, time_to_see_data_ms from metrics_time_to_see_data where query_id = '50c8bde3-5a9c-41ea-a5c8-f1c134ce721c'`

I bet that table isn't _query_ aware, and needs a proper setup. We didn't properly create it IIRC

@yakkomajuri might have more context

**Comment by mariusandra** [comment]
This is less about timing the queries, and more about capturing the query JSON as an event, so we could do analysis on the types of queries people run :).

**Comment by pauldambra** [comment]
We capture those to the query log today `select JSONExtractString(log_comment, 'query') from system.query_log where JSONHas(log_comment, 'query') limit 5`

That might not be the best place though 🤷

**Comment by mariusandra** [comment]
The system query log is great, but it's not useful if I want to make non-metabase dashboards aka dogfooding :).

File: frontend/src/queries/query.ts [hunk 1]
```diff
@@ -113,44 +114,58 @@ export async function query<N extends DataNode = DataNode>(
...
-                sessionId: queryNode.sessionId ?? currentSessionId(),
-                sessionStart: queryNode.sessionStart ?? now().subtract(1, 'day').toISOString(),
-                sessionEnd: queryNode.sessionEnd ?? now().toISOString(),
-            },
-            methodOptions
-        )
+        posthog.capture('Query', { query: queryNode, duration: performance.now() - startTime, ...logParams })
+        return response
+    } catch (e) {
+        posthog.capture('Query Error', { query: queryNode, duration: performance.now() - startTime, ...logParams })
```
Comment (Twixes) [inline frontend/src/queries/query.ts:166]: Nit: I think it's quite rare for our events to look like "Title Case", they're usually more something along the lines of `query completed` or `query failed`

**Review by Twixes — COMMENTED** [review]
Makes sense to me, though I have a small comment on event naming.
BTW what's the plan with time to see data? That kind of seems redundant and not dogfooding-friendly

File: frontend/src/queries/query.ts (hunk 1 above)
Comment (mariusandra) [reply frontend/src/queries/query.ts:166]: makes sense, renamed

**Comment by mariusandra** [comment]
Thanks for the review!

Re: time to see data: 🤷. Here I'm less interested in the "time to see", but just the who and the what.

**Comment by mariusandra** [comment]
I'm still not sure if there's anything blocking this PR...? Is there? 🤔

**Review by Twixes — APPROVED** [review]
Looks good

**Pull request merged**